src/hangar/checkout.py

```diff
@@ -532,6 +532,13 @@ def commit(self, commit_message):
         '''
         self.__acquire_writer_lock()
         logger.info(f'Commit operation requested with message: {commit_message}')
+        metadataconman = self._metadata._is_conman
+        if metadataconman:
```

Comment:
You might need to do some additional checks on the transaction register here. While normally each teardown of the metadata and dataset class instances will either abort or commit pending reader transactions from their threads, you need to consider the state of all of them and not just one. 

The transaction register is just a reference counting scheme which shares transaction handles between multiple concurrent requesters 
 When you're operating within a context manager at the checkout level you'll have to care  that each outstanding transaction handle has actually been closed. I don't know if it's a concern in practice, but this is an area where you want to make sure that all writes have been actually flushed to disk

Does that make sense?

Reply:
@rlizzo I can't think of an instance this is going to break. So leaving it as it is. Let me know if it rings a bell for you.

Reply:
See my comment https://github.com/tensorwerk/hangar-py/pull/94#issuecomment-513514480 below. I'm good with this!